flycheck.el

```diff
@@ -3942,8 +3967,10 @@ Return a list with the contents of the table cell."
                    column 'flycheck-error-list-column-number)
                   (flycheck-error-list-make-cell
                    (symbol-name (flycheck-error-level error)) level-face)
-                  (flycheck-error-list-make-cell
-                   id-str 'flycheck-error-list-id id-str)
+                  ;; Error ID has a button to explain the error
+                  (list id-str
+                        'type 'flycheck-error-list-explain-error
```

Comment:
Also, the presence of an explainer does not necessarily means there will be an explanation (the explainer may return nil).  In that case, checking the presence of the *explanation* might be costly as it will most probably call external processes (it does for rust).